src/extension.ts

```diff
@@ -687,6 +687,9 @@ export class Extension implements RunHooks {
             }
           });
         }
+
+        if (completedDecorations.length > 1000) // too many decorations slow down the editor
```

Comment:
Perhaps a 100? I can't imagine anyone reading more than five :)

Reply:
This limit is per-file, so 100 will be used up quickly. And we don't know which ones the user wants to read. So i'd stay high enough that we only prevent the performance bottleneck, and nothing else.